inspirehep/modules/disambiguation/core/ml/models.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+
+"""Disambiguation core ML models."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import pickle
+
+import numpy as np
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.pipeline import Pipeline
+from sklearn.svm import LinearSVC
+
+from beard.utils import normalize_name
+
+
+class EthnicityEstimator(object):
+    def __init__(self, C=4.0):
+        self.C = C
+
+    def load(self, input_filename):
+        ethnicities, lasts, firsts = [], [], []
+        with open(input_filename, 'r') as fd:
+            reader = csv.DictReader(fd)
+            for row in reader:
+                ethnicities.append(row['RACE'])
+                lasts.append(row['NAMELAST'])
+                firsts.append(row['NAMEFRST'])
```

Comment:
This is a little more convoluted and slow than the original https://github.com/inspirehep/beard/blob/87b37a3028825b5199616b195663fee526c26be3/examples/applications/author-disambiguation/ethnicity.py#L39, but it has the benefit of avoiding a big dependency such as `pandas` for exactly one line of code.